packages/editor/src/components/post-text-editor/index.js

```diff
@@ -13,67 +13,59 @@ import { parse } from '@wordpress/blocks';
 import { withSelect, withDispatch } from '@wordpress/data';
 import { withInstanceId, compose } from '@wordpress/compose';
 
-/**
- * Returns the PostTextEditor state given a set of props.
- *
- * @param {Object} props Component props.
- *
- * @return {Object} State object.
- */
-function computeDerivedState( props ) {
-	return {
-		persistedValue: props.value,
-		value: props.value,
-		isDirty: false,
-	};
-}
-
-class PostTextEditor extends Component {
+export class PostTextEditor extends Component {
 	constructor() {
 		super( ...arguments );
 
-		this.startEditing = this.startEditing.bind( this );
 		this.edit = this.edit.bind( this );
 		this.stopEditing = this.stopEditing.bind( this );
 
-		this.state = {
-			value: null,
-			isDirty: false,
-		};
+		this.state = {};
 	}
 
-	static getDerivedPropsFromState( props, state ) {
-		// If we receive a new value while we're editing (but before we've made
-		// changes), go ahead and clobber the local state
-		if ( state.persistedValue !== props.value && ! state.isDirty ) {
-			return computeDerivedState( props );
+	static getDerivedStateFromProps( props, state ) {
+		if ( state.isDirty ) {
+			return null;
 		}
 
-		return null;
-	}
-
-	startEditing() {
-		// Copying the post content into local state ensures that edits won't be
-		// clobbered by changes to global editor state
-		this.setState( { value: this.props.value } );
+		return {
+			value: props.value,
+			isDirty: false,
+		};
 	}
 
+	/**
+	 * Handles a textarea change event to notify the onChange prop callback and
+	 * reflect the new value in the component's own state. This marks the start
+	 * of the user's edits, if not already changed, preventing future props
+	 * changes to value from replacing the rendered value. This is expected to
+	 * be followed by a reset to dirty state via `stopEditing`.
+	 *
+	 * @see stopEditing
+	 *
+	 * @param {Event} event Change event.
+	 */
 	edit( event ) {
 		const value = event.target.value;
 		this.props.onChange( value );
```

Comment:
So this is a bit weird :). I suppose this is only needed to trigger the post dirtiness, right? I wonder if a specific action is not better.

Reply:
Granted it's not being changed in this PR

Reply:
Can you elaborate on what's weird here? `onChange` is called to surface the edit to the editor store immediately (so as to mark for dirty detection).

Reply:
Why are we filling the `content` in the `edits`? For me the `blocks` is the `edits` of the `content` property. And we're already filling the blocks properly when persisting. I bet this is only needed to trigger a "dirty" state.

Reply:
>I bet this is only needed to trigger a "dirty" state.

Pretty much this, yes 😄

Reply:
Is this better than a `SET_DIRTY` action or something like that? It's a bit clearer